### ISSUE TITLE
Can't set custom HostnameVerifier on DropWizard client

### ISSUE THREAD

**Issue by tbartley** [dropwizard/dropwizard#1663]
While the improvements to TLS configuration of HTTP clients in 1.0.0 (maybe prior) are awesome, as part of that process the ability to set a custom HostnameVerifier easily on the HTTP client has been lost.

You used to be able to do e.g. as:

```
JerseyClientConfiguration myJerseyClientConfiguration = <some configuration>;
HostnameVerifier verifier = new MyCustomHostnameVerifier();
JerseyClientBuilder clientBuilder = new JerseyClientBuilder(env);
clientBuilder.using(myJerseyClientConfiguration).using(verifier);
Client httpClient = clientBuilder.build();
```
Same is true for `HttpClientBuilder` too.

You can still do it by creating a custom Apache `Registry<ConnectionSocketFactory>` but you need to set up socket factories for every scheme. Pretty heavyweight.

I have a pull request that I'll submit and reference this issue.

**Comment by tbartley** [comment]
Submitted #1664 pull against v1.0.x branch but it fails VersionEye testing. I didn't change any POMs - how can I fix that?

**Comment by arteam** [comment]
Hi, @tbartley. It's okay, the maintainers can test the pull request locally. I will review it shortly.

Could you tell, from which Dropwizard's version you are migrating to 1.0.0? I don't remember we provided the ability to set a custom `HostnameVerifier` in previous versions. I think we always exposed only a registry of socket factories. If it's a regression, we could make a bug-fix release for 1.0.*, otherwise I more fond to push the change to main branch which will be 1.1.0.

**Comment by tbartley** [comment]
We are coming all the way from 0.7.2 😊 . If only the software didn't work so well we'd've been more motivated to move sooner.
